django/db/backends/base/base.py

```diff
@@ -641,3 +648,21 @@ def copy(self, alias=None, allow_thread_sharing=None):
         if allow_thread_sharing is None:
             allow_thread_sharing = self.allow_thread_sharing
         return type(self)(settings_dict, alias, allow_thread_sharing)
+
+    def _push_execute_hook(self, hook, for_many=None):
+        self.execute_hooks.append((hook, for_many))
+
+    def _pop_execute_hook(self):
+        return self.execute_hooks.pop()
+
+    @contextmanager
+    def add_execute_hook(self, hook, for_many=None):
+        self._push_execute_hook(hook, for_many)
+        yield
+        self._pop_execute_hook()
+
+    def get_execute_hooks(self, for_many):
+        return [
+            hook for (hook, hook_for_many) in self.execute_hooks
+            if hook_for_many is None or hook_for_many == for_many
```

Comment:
this `is None` logic is untested and unused, remove it and just do the `==` comparison?

Reply:
Oh, but  it is -- `hook_for_many` here is the hook's `for_many` flag, the parameter coming from `add_execute_hook()` (now renamed `exectute_hook()`), whose default value is `None`. 

`for_many` here represents the called function -- I guess I should clarify this in a comment.

Reply:
Thanks for the extra explanation, I understand now